lib/EnsEMBL/REST/Controller/Homology.pm

```diff
@@ -50,7 +50,9 @@ sub get_adaptors :Private {
 
   try {
     my $species = $c->stash()->{species};
-    my $compara_dba = $c->model('Registry')->get_best_compara_DBAdaptor($species, $c->request()->param('compara'), $self->default_compara());
+    my $param_compara = $c->request()->param('compara');
```

Comment:
This change does not seem to add anything useful, except some improved readability.

Reply:
No, it is more than that. In fact it fixes a bug.
The arguments for a subroutine call in Perl are passed as a list. The second argument to get_best_compara_DBAdaptor here is $c->request()->param('compara') which will be evaluated in a LIST context.
The problem is that in LIST context, the Catalyst::Request returns an empty list for an argument that does not exist:
https://github.com/perl-catalyst/catalyst-runtime/blob/master/lib/Catalyst/Request.pm#L783
So when the request url does not have 'compara=' parameter, the arguments list passed to get_best_compara_DBAdaptor here is [$species, (), $default_compara] which is [$species, $default_compara], causing second argument to be skipped.
The change here is to evaluate the Catalyst::Request::param call in a SCALAR context so that it returns 'undef' for a parameter that does not exist.

Reply:
Thank you for the explanation, we've learned something more about Catalyst :+1:, at least me.
Might have been good to have it explained at PR creation time, thanks anyway.